parquet-hadoop/src/main/java/org/apache/parquet/crypto/CryptoPropertiesFactory.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.crypto;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.hadoop.BadConfigurationException;
+import org.apache.parquet.hadoop.api.WriteSupport.WriteContext;
+import org.apache.parquet.hadoop.util.ConfigurationUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * CryptoPropertiesFactory interface enables transparent activation of Parquet encryption.
+ *
+ * It's customized implementations produce encryption and decryption properties for each Parquet file, using the information
+ * available in Parquet file writers and readers: file path, file extended schema (in writers only) - and also Hadoop
+ * configuration properties that can pass custom parameters required by a crypto factory. A factory implementation can
+ * use or ignore any of these parameters.
```

Comment:
```suggestion
 * use or ignore any of these inputs.
```
was a bit misleading, my fault.